### PR TITLE
feat: enable --iam-service options for including IAM methods

### LINKED ISSUE #315: support `reroute_to_grpc_interface` for extra needed service
For `@google-cloud/kms`, `SetIamPolicy` and `GetIamPolicy` is defined in `googleapis/googleapis/google/cloud/kms/v1/cloudkms_gapic.yaml` and it reroutes to `google.iam.v1.IAMPolicy` which is not supported by micro-generator now.
```
reroute_to_grpc_interface: google.iam.v1.IAMPolicy
```
For go, they add the rerouting methods manually to generated client code, [but in separate directory, or it will be rewritten by synthtool]. For current Typescript library, we manually create IAM client, and then mix-in IAM and KMS clients together(in synth.py).

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
fix #315

### CODE REVIEW

File: baselines/iam-service-test/src/v1/key_management_service_client.ts.baseline [hunk 1]
```diff
@@ -0,0 +1,2583 @@
...
+    const callSettings = new gax.CallSettings(options);
+    this.initialize();
+    return this.descriptors.page.listImportJobs.asyncIterate(
+      this.innerApiCalls['listImportJobs'] as GaxCall,
+      request as unknown as RequestType,
+      callSettings
+    ) as AsyncIterable<protos.google.cloud.kms.v1.IImportJob>;
+  }
+/**
+ * This part will be added into src/version/client.ts.
```
Comment (alexander-fenster) [inline baselines/iam-service-test/src/v1/key_management_service_client.ts.baseline:2539]: I don't think this comment should get into the generated code. If it's supposed to be a template comment, use `{#  #}`.

File: baselines/iam-service-test/test/gapic_key_management_service_v1.ts.baseline [hunk 2]
```diff
@@ -0,0 +1,2657 @@
...
+                (client.descriptors.page.listImportJobs.asyncIterate as SinonStub)
+                    .getCall(0).args[1], request);
+            assert.strictEqual(
+                (client.descriptors.page.listImportJobs.asyncIterate as SinonStub)
+                    .getCall(0).args[2].otherArgs.headers['x-goog-request-params'],
+                expectedHeaderRequestParams
+            );
+        });
+    });
+});
```
Comment (alexander-fenster) [inline baselines/iam-service-test/test/gapic_key_management_service_v1.ts.baseline:2657]: IAM methods should be covered by the generated unit tests (both callback and `await` use cases, both error and no-error cases, similarly to how it's done for regular simple methods).

File: typescript/src/generator.ts [hunk 3]
```diff
@@ -110,6 +111,14 @@ export class Generator {
       this.bundleConfigs = new BundleConfigClient().fromObject(info);
     }
   }
+  private readIamService() {
```
Comment (alexander-fenster) [inline typescript/src/generator.ts:114]: nit: add an empty line before this method please :)

File: typescript/src/schema/api.ts [hunk 4]
```diff
@@ -74,7 +74,8 @@ export class API {
           options.grpcServiceConfig,
           allResourceDatabase,
           resourceDatabase,
-          options.bundleConfigs
+          options.bundleConfigs,
+          options.iamService
```
Comment (alexander-fenster) [inline typescript/src/schema/api.ts:78]: This is going out of control here. Let's stop this now :)  Can I ask you to export an `interface ProtoParameters` from `proto.ts` and make the `Proto` constructor accept one object?

```ts
map[fd.name!] = new Proto({
  fd,
  packageName,
  grpcServiceConfig: options.grpcServiceConfig,
  ...
});
```

File: typescript/src/schema/proto.ts [hunk 5]
```diff
@@ -612,7 +616,8 @@ export class Proto {
           grpcServiceConfig,
           allResourceDatabase,
           resourceDatabase,
-          bundleConfigs
+          bundleConfigs,
+          iamService
```
Comment (alexander-fenster) [inline typescript/src/schema/proto.ts:620]: Same here: it's too many parameters for one function already.

File: baselines/iam-service-test/src/v1/key_management_service_client.ts.baseline (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply baselines/iam-service-test/src/v1/key_management_service_client.ts.baseline:2539]: sure, moved it!

File: typescript/src/generator.ts (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply typescript/src/generator.ts:114]: done. Thank you!

File: typescript/src/schema/api.ts (hunk 4 above)
Comment (xiaozhenliu-gg5) [reply typescript/src/schema/api.ts:78]: sure! we can reuse the `naming.Options` interface. Thanks! same below.

File: baselines/iam-service-test/test/gapic_key_management_service_v1.ts.baseline (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply baselines/iam-service-test/test/gapic_key_management_service_v1.ts.baseline:2657]: Sure, adding now.

Comment (xiaozhenliu-gg5) [reply baselines/iam-service-test/test/gapic_key_management_service_v1.ts.baseline:2933]: done.

File: templates/typescript_gapic/src/$version/$service_client.ts.njk [hunk 6]
```diff
@@ -642,6 +649,9 @@ export class {{ service.name }}Client {
     ) as AsyncIterable<{{ util.toInterface(method.pagingResponseType) }}>;
   }
 {%- endfor %}
+{%- if service.iamService %}
+{{ util.iamServiceMethods()}}
```
Comment (alexander-fenster) [inline templates/typescript_gapic/src/$version/$service_client.ts.njk:653]: nit: `util` is really overloaded with all the kinds of random stuff. Maybe make `_iam.njk` (starting with underscore so it gets ignored by the templater) and put both service methods and tests there?

Comment (xiaozhenliu-gg5) [reply templates/typescript_gapic/src/$version/$service_client.ts.njk:653]: got it, it makes perfect sense to move them to another njk file. Thank you for the suggestion!

Comment (xiaozhenliu-gg5) [reply templates/typescript_gapic/src/$version/$service_client.ts.njk]: done.

**Pull request merged**